neps/nep-0491.md

```diff
@@ -0,0 +1,359 @@
+---
+NEP: 491
+Title: Non-Refundable Storage Staking
+Authors: Jakob Meier <jakob@near.org>
+Status: Draft
+DiscussionsTo: https://gov.near.org/t/proposal-locking-account-storage-refunds-to-avoid-faucet-draining-attacks/34155
+Type: Protocol Track
+Version: 1.0.0
+Created: 2023-07-24
+LastUpdated: 2023-07-26
+---
+
+
+## Summary
+
+Non-refundable storage allows to create accounts with arbitrary state for users,
+without being susceptible to refund abuse.
+
+This is done by tracking non-refundable balance in a separate field of the
+account. This balance is only useful for storage staking and otherwise can be
+considered burned.
+
+
+## Motivation
+
+Creating new accounts on chain costs a gas fee and a storage staking fee. The
+more state is added to the account, the higher the storage staking fees. When
+deploying a contract on the account, it can quickly go above 1 NEAR per account.
+
+Some business models are okay with paying that fee for users upfront, just to
+get them onboarded. However, if a business does that today, their users can
+delete their new accounts and spend the tokens intended for storage staking in
+other ways. Since this is free for the user, they are financially incentivized
+to repeat this action for as long as the business has funds left in the faucet.
+
+The protocol should allow to create accounts in a way that is not susceptible to
+such refund abuse. This would at least change the incentives such that creating
+fake users is no longer profitable.
+
+Non-refundable storage staking is a further improvement over
+[NEP-448](https://github.com/near/NEPs/pull/448) (Zero Balance Accounts) which
+addressed the same issue but is limited to 770 bytes per account. By lifting the
+limit, sponsored accounts can be used in combination with smart contracts.
```

Comment:
Perhaps the following part could use a bit more details (expanded on, in some section below):

> By lifting the limit

with respect to how it interacts with "free 770 bytes per account" concept. Because strictly speaking, the way I understand this prop, it's not exactly a generalization of https://github.com/near/NEPs/pull/448 but rather a standalone feature that extends it conceptually.

It seems both features interplay nicely together, just want to make sure it doesn't create unnecessary burden for developers when they decide which one to go with.